### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by awls99**
This adds the possibility of calling integration tests in the pipeline

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ on:
         required: false
         type: string
       node-version:
-        default: "14.x"
+        default: "18.x"
```
Comment (BrianArch96) [inline .github/workflows/deploy.yaml:25]: i can't remember why but I believe @cbarlow1993 is wary of bumping to v18 in some places, gonna tag him so hes aware of this.

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -93,6 +93,12 @@ jobs:
           heap: ${{ inputs.heap-size }}
           region: ${{ inputs.region }}
           GH_TOKEN: ${{ secrets.GH_TOKEN_SEMANTIC_RELEASE_AND_NPM_REGISTRY }}
+      - name: Integration Tests
+        if: ${{ inputs.github-stage != 'temporary' }}      
+        run: npm run test:integration
```
Comment (BrianArch96) [inline .github/workflows/deploy.yaml:98]: is this replacing what we've previously used as `test:stage`?

**Review by BrianArch96 — CHANGES_REQUESTED** [review]
just a small question

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (awls99) [reply .github/workflows/deploy.yaml:98]: This is not replacing, there's a new thing we can use

Comment (BrianArch96) [reply .github/workflows/deploy.yaml:98]: so what is the difference between this and `test:stage`? As that is our current command for calling integration tests

Comment (awls99) [reply .github/workflows/deploy.yaml:98]: Two things: 
one `test:integration` is a way better name (I was genuinely convinced that `test:stage` was testing staging environment)
And, jokes aside, this just makes it so integration tests run after the deploy.
If we're attached to the `test:stage` name 🤷 we can keep it, but I've been seeing `test:integration` on newer PR's

**Review by BrianArch96 — APPROVED** [review]

**Pull request merged**